ibis/pandas/execution/tests/test_timecontext.py

```diff
@@ -0,0 +1,100 @@
+import pandas as pd
+import pandas.util.testing as tm
+import pytest
+
+import ibis
+import ibis.common.exceptions as com
+
+pytestmark = pytest.mark.pandas
+
+
+def test_execute_with_timecontext(time_table):
+    expr = time_table
+    # define a time context for time-series data
+    context = ('20170101', '20170103')
+
+    # without time context, execute produces every row
+    df_all = expr.execute()
+    assert len(df_all['time']) == 8
+
+    # with context set, execute produces only rows within context
+    df_within_context = expr.execute(timecontext=context)
+    assert len(df_within_context['time']) == 1
+
+
+merge_asof_minversion = pytest.mark.skipif(
+    pd.__version__ < '0.19.2',
+    reason='at least pandas-0.19.2 required for merge_asof',
+)
+
+
+def test_bad_timecontext(time_table):
+    expr = time_table
+    # define context with illegal string
+    with pytest.raises(ValueError):
+        context = ('bad', 'context')
+        expr.execute(timecontext=context)
+    # define context with unsupport type int
+    with pytest.raises(com.IbisError):
+        context = (20091010, 20100101)
+        expr.execute(timecontext=context)
+    # define context with too many values
+    with pytest.raises(com.IbisError):
+        context = ('20101010', '20100101', '20101010')
+        expr.execute(timecontext=context)
+    # define context with too few values
+    with pytest.raises(com.IbisError):
+        context = pd.Timestamp('20101010')
```

Comment:
what happens if no TIME_COL?

Reply:
It will raise an IbisError as well, just added a new test case to catch that.